core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,255 @@
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.Type;
+import org.elasticsearch.index.search.geo.InMemoryGeoBoundingBoxQuery;
+import org.elasticsearch.test.geo.RandomShapeGenerator;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class GeoBoundingBoxQueryBuilderTest extends BaseQueryTestCase<GeoBoundingBoxQueryBuilder> {
+    @Override
+    protected GeoBoundingBoxQueryBuilder doCreateTestQueryBuilder() {
+        GeoBoundingBoxQueryBuilder builder = new GeoBoundingBoxQueryBuilder("mapped_geo");
+        Rectangle box = RandomShapeGenerator.xRandomRectangle(getRandom(), RandomShapeGenerator.xRandomPoint(getRandom()));
+
+        if (frequently()) {
+            // check the top-left/bottom-right combination of setters
+            int path = randomIntBetween(0, 2);
+            switch(path) {
+                case 0: 
+                    builder.topLeft(new GeoPoint(box.getMaxY(), box.getMinX()));
+                    break;
+                case 1:
+                    builder.topLeft(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMinX()));
+                    break;
+                default:
+                    builder.topLeft(box.getMaxY(), box.getMinX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch(path) {
+            case 0: 
+                builder.bottomRight(new GeoPoint(box.getMinY(), box.getMaxX()));
+                break;
+            case 1:
+                builder.bottomRight(GeohashUtils.encodeLatLon(box.getMinY(), box.getMaxX()));
+            default:
+                builder.bottomRight(box.getMinY(), box.getMaxX());
+            }
+        } else {
+            // check the deprecated top-right/bottom-left combination of setters
+            int path = randomIntBetween(0, 2);
+            switch(path) {
+                case 0: 
+                    builder.topRight(new GeoPoint(box.getMaxY(), box.getMaxX()));
+                    break;
+                case 1:
+                    builder.topRight(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMaxX()));
+                    break;
+                default:
+                    builder.topRight(box.getMaxY(), box.getMaxX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch(path) {
+            case 0: 
+                builder.bottomLeft(new GeoPoint(box.getMinY(), box.getMinX()));
+                break;
+            case 1:
+                builder.bottomLeft(GeohashUtils.encodeLatLon(box.getMinY(), box.getMinX()));
+            default:
+                builder.bottomLeft(box.getMinY(), box.getMinX());
+            }
+        }
+
+        if (randomBoolean()) {
+            builder.normalize(randomBoolean());
+        }
+
+        // TODO we aren't testing the "memory" scenario - needs a node pulled up for testing
+        builder.type(Type.INDEXED);
```

Comment:
especially given the recent build changes, we have to do everything we can to not move to requiring nodes. That said do you think it would be crazy to mock `IndexFieldDataService` and all of the needed field data aspects so we can test memory type too?